src/state/opamEnv.ml

```diff
@@ -688,18 +688,18 @@ let updates_nix st =
      | Some env -> List.map resolve_separator_and_format env)
   | _ -> []
 
-let updates ~set_opamroot ~set_opamswitch ?force_path st =
+let updates ~set_opamroot ~set_opamswitch ?force_path ~build_env st =
   let common =
     updates_common ~set_opamroot ~set_opamswitch st.switch_global.root st.switch
   in
-  common @ compute_updates ?force_path st @ updates_nix st
+  compute_updates ?force_path ~build_env st @ common @ updates_nix st
```

Comment:
common and nix updates should probably be moved in `compute_updates` to centralise the ordering in one function. I also don't think nix env should be able to take priority over every others